### DISCUSSION TITLE
Mqtt setup for esphome pugin

### DISCUSSION THREAD

**Discussion by sat-fan2019** [jomjol/AI-on-the-edge-device#1362] [Q&A]
I am working for days at that project.
At least managed to get the values ​​read correctly.
Only the integration into HA does not want to work. Can’t find any useful instructions.
If I create a device for the ESPHome and then write the config to the ESP via the Ras4, I can no longer get to the UI via the wlan 

Afterwords Esp-Home device is found for this, but without an entity!
Is the procedure correct at all to run an ESP Home install over it? 

Looks like, that the ESPHome setup overwrites my installation !

I have this entry in the config:

[MQTT]
Url = mqtt://10.0.0.88:1883
MainTopic = water meter
ClientID = water
user = wateruser
password = waterfall

I then created this user in Mosquitto Broker… Is that even true? 
I’m afraid I didn’t understand the principle of how to get an entity from ESP as a sensor in HA.
thank you for your help

**Comment by caco3** [comment]
I never used ESPHome, so don't know if and how it is supposed to work with that.

The installation  of AI-on-the-edge-device is documented in https://github.com/jomjol/AI-on-the-edge-device/wiki/Installation#firmware-flashing

MQTT for Homeassistant is documented in https://github.com/jomjol/AI-on-the-edge-device/wiki/Integration-Home-Assistant
You can use the latest rolling since the last release does not support Homeassistant Discovery yet.

Latest rolling: https://github.com/jomjol/AI-on-the-edge-device/actions?query=branch%3Arolling

**Comment by DanielWe2** [comment]
Don't use ESP Home.  ESPHome generates a total different firmware for the ESP overriding the AI on the edge firmware.

Just configure a few MQTT sensors in Home Assistant (see links from caco3).  
(you first have to configure a connection to a MQTT broker in the AI on the edge web ui. If you do not have a MQTT broker you can install one in Home Assistant )
